test/e2e/exposition_test.go

```diff
@@ -0,0 +1,251 @@
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			// Update configmap
+			cm, err := clients.KubeClient.CoreV1().ConfigMaps(namespace).Get(context.Background(), config.IstioConfigName, metav1.GetOptions{})
+			if err != nil {
+				t.Fatalf("Failed to get configmap %s/%s", namespace, config.IstioConfigName)
+			}
+
+			cm.Data = c.configMapData
+
+			if _, err := clients.KubeClient.CoreV1().ConfigMaps(namespace).Update(context.Background(), cm, metav1.UpdateOptions{}); err != nil {
+				t.Fatalf("Failed to update configmap %s/%s: %v", namespace, config.IstioConfigName, err)
+			}
+
+			// Create ingress
+			name, port, cancel := ingress.CreateRuntimeService(context.Background(), t, clients.NetworkingClient, networking.ServicePortNameHTTP1)
+			hosts := []string{name + ".example.com"}
+
+			spec := v1alpha1.IngressSpec{
+				Rules: []v1alpha1.IngressRule{{
+					Hosts:      hosts,
+					Visibility: v1alpha1.IngressVisibilityExternalIP,
+					HTTP: &v1alpha1.HTTPIngressRuleValue{
+						Paths: []v1alpha1.HTTPIngressPath{{
+							Splits: []v1alpha1.IngressBackendSplit{{
+								IngressBackend: v1alpha1.IngressBackend{
+									ServiceName:      name,
+									ServiceNamespace: test.ServingNamespace,
+									ServicePort:      intstr.FromInt(port),
+								},
+							}},
+						}},
+					},
+				}},
+			}
+
+			ing, cancel2 := ingress.CreateIngress(context.Background(), t, clients.NetworkingClient, spec, func(target *v1alpha1.Ingress) {
+				if target.ObjectMeta.Labels == nil {
+					target.ObjectMeta.Labels = make(map[string]string)
+				}
+
+				for k, v := range c.ingressLabels {
+					target.ObjectMeta.Labels[k] = v
+				}
+			})
+
+			// Wait ingress to be ready and get its status
+			for {
+				ing, err = clients.NetworkingClient.NetworkingClient.Ingresses.Get(context.Background(), ing.Name, metav1.GetOptions{})
+				if err != nil {
+					t.Fatalf("Failed to get ingress %s", ing.Name)
+				}
+
+				if ing.IsReady() {
+					break
+				}
+			}
+
+			// Check
+			diff := cmp.Diff(c.expectedPrivateIngresses, ing.Status.PrivateLoadBalancer.Ingress)
+			if diff != "" {
+				t.Error("Unexpected private ingresses (-want, +got):", diff)
+			}
+
+			diff = cmp.Diff(c.expectedPublicIngresses, ing.Status.PublicLoadBalancer.Ingress)
+			if diff != "" {
+				t.Error("Unexpected public ingresses (-want, +got):", diff)
+			}
+
+			cancel2()
+			cancel()
+
+			// Restore configmap
+			cm, err = clients.KubeClient.CoreV1().ConfigMaps(namespace).Get(context.Background(), config.IstioConfigName, metav1.GetOptions{})
+			if err != nil {
+				t.Fatalf("Failed to get configmap %s/%s", namespace, config.IstioConfigName)
+			}
+
+			cm.Data = oldConfig.Data
+
+			if _, err := clients.KubeClient.CoreV1().ConfigMaps(namespace).Update(context.Background(), cm, metav1.UpdateOptions{}); err != nil {
+				t.Fatalf("Failed to restore configmap %s/%s: %v", namespace, config.IstioConfigName, err)
+			}
+		})
+	}
+}
+
+func replaceTabs(s string) string {
```

Comment:
We could have as an alternative:

```
func toYamlNewFormatGateways(data []Gateway) string {
	bytes, _ := yaml.Marshal(data)
	return string(bytes)
}

"external-gateways": toYamlNewFormatGateways([]Gateway{
  {
    Namespace:  "unused",
    Name:       "unused",
    ServiceURL: "default.default.svc.cluster.local",
  },
  {
    Namespace:  "namespace",
    Name:       "gateway",
    ServiceURL: "istio-gateway.istio-system.svc.cluster.local",
    LabelSelector: &metav1.LabelSelector{
      MatchExpressions: []metav1.LabelSelectorRequirement{
        {
          Key:      "key",
          Operator: metav1.LabelSelectorOpIn,
          Values:   []string{"value"},
        },
      },
    },
  }},
),
```

Reply:
I think I rather have a string that reflects exactly what a user would have written, rather than generate it

Reply:
For the scenarios you want to check just the user syntax yes, for the rest where you want to run test cases based on valid content it is safer. Anyway no strong preference.